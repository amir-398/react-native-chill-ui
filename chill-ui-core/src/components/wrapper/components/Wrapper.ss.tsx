import type { WrapperPropsSs } from '@types';
import type { PropsWithChildren } from 'react';

import { BoxSs } from '@components/box';

import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView';

/**
 * The `<Wrapper />` component provides a flexible container with default styling and SafeAreaView support.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { Wrapper } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <Wrapper style={{ backgroundColor: '#f0f0f0' }}>
 *   <String>Content</String>
 * </Wrapper>
 * ```
 *
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function Wrapper(props: PropsWithChildren<WrapperPropsSs>) {
  const { children, edges, fill, grow, hasSafeArea, px, style } = props;

  const content = <BoxSs style={[wrapperSv({ fill, grow, px }), style]}>{children}</BoxSs>;

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

Wrapper.displayName = 'Wrapper';
