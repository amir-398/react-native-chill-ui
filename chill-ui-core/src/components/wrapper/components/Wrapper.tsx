import type { WrapperPropsTw } from '@types';
import type { PropsWithChildren } from 'react';

import { Box } from '@components/box';
import { cn, classNamePropsHandler, classNameHandler, styleHandler } from '@utils';

import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView';

/**
 * The `<Wrapper />` component provides a flexible container with default styling and SafeAreaView support.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { Wrapper } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <Wrapper>
 *   <String>Content</String>
 * </Wrapper>
 * ```
 * @param className - Custom CSS classes for the wrapper (NativeWind only)
 * @param edges - Safe area edges to apply when hasSafeArea is true
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function Wrapper(props: PropsWithChildren<WrapperPropsTw>) {
  classNamePropsHandler(props, 'Wrapper');

  const { children, className, edges, fill, grow, hasSafeArea, px, style, ...rest } = props;

  const content = (
    <Box
      {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
      {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
      {...rest}
    >
      {children}
    </Box>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

Wrapper.displayName = 'Wrapper';
