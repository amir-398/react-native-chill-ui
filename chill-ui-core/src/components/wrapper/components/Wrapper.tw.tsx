import type { WrapperPropsTw } from '@types';
import type { PropsWithChildren } from 'react';

import { cn } from '@utils';
import { BoxTw } from '@components/box';

import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.tw';

/**
 * The `<Wrapper />` component provides a flexible container with default styling and SafeAreaView support.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { Wrapper } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <Wrapper className="bg-gray-100 p-4">
 *   <String>Content</String>
 * </Wrapper>
 * ```
 *
 * @param className - Custom CSS classes for the wrapper (NativeWind)
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function Wrapper(props: PropsWithChildren<WrapperPropsTw>) {
  const { children, className, edges, fill, grow, hasSafeArea, px, ...rest } = props;

  const content = (
    <BoxTw className={cn(wrapperTv({ fill, grow, px }), className)} {...rest}>
      {children}
    </BoxTw>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

Wrapper.displayName = 'Wrapper';
