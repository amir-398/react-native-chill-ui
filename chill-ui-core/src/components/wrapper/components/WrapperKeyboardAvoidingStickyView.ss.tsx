import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingStickyViewPropsSs } from '@types';

import { customConsole } from '@utils';

import { Wrapper } from './Wrapper.ss';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.ss';

// Optional import with error handling
let KeyboardStickyView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardStickyView = keyboardController.KeyboardStickyView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * KeyboardAvoidingView wrapper component for keyboard avoidance.
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingStickyView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingStickyView>
 *
 * // With SafeAreaView
 * <WrapperKeyboardAvoidingStickyView hasSafeArea edges={['top', 'bottom']}>
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingStickyView>
 * ```
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param px - Padding for the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param edges - Safe area edges to apply when hasSafeArea is true
 * @param style - Style prop
 * @param enabled - Whether the keyboard avoiding view is enabled
 * @param offset - Offset for the keyboard avoiding sticky view { close: number; open: number }
 * @param children - Child components to render
 */
export function WrapperKeyboardAvoidingStickyView(props: PropsWithChildren<WrapperKeyboardAvoidingStickyViewPropsSs>) {
  const { children, edges, fill, grow, hasSafeArea, offset, px, style, ...rest } = props;

  if (!KeyboardStickyView) {
    return (
      <Wrapper style={[wrapperSv({ fill, grow, px }), style]} hasSafeArea={hasSafeArea} edges={edges} {...rest}>
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardStickyView style={[wrapperSv({ fill, grow, px }), style]} offset={offset} {...rest}>
      {children}
    </KeyboardStickyView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingStickyView.displayName = 'WrapperKeyboardAvoidingStickyView';
