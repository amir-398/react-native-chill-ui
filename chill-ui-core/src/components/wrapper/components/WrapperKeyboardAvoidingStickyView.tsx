import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingStickyViewPropsTw } from '@types';

import { classNameHandler, cn, customConsole, styleHandler } from '@utils';

import { Wrapper } from './Wrapper.tw';
import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.tw';

// Optional import with error handling
let KeyboardStickyView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardStickyView = keyboardController.KeyboardStickyView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * The `<WrapperKeyboardAvoidingStickyView />` component provides a KeyboardStickyView wrapper for keyboard avoidance with sticky behavior.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { WrapperKeyboardAvoidingStickyView } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingStickyView offset={{ close: 0, open: 20 }}>
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingStickyView>
 * ```
 *
 * @param className - Custom CSS classes for the wrapper (NativeWind only)
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param enabled - Whether the keyboard avoiding sticky view is enabled (default: `true`)
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param offset - Offset for the keyboard avoiding sticky view: `{ close: number; open: number }`
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function WrapperKeyboardAvoidingStickyView(props: PropsWithChildren<WrapperKeyboardAvoidingStickyViewPropsTw>) {
  const { children, className, edges, fill, grow, hasSafeArea, px, style, ...rest } = props;

  if (!KeyboardStickyView) {
    return (
      <Wrapper
        {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
        {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
        hasSafeArea={hasSafeArea}
        edges={edges}
        {...rest}
      >
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardStickyView
      {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
      {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
      {...rest}
    >
      {children}
    </KeyboardStickyView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingStickyView.displayName = 'WrapperKeyboardAvoidingStickyView';
