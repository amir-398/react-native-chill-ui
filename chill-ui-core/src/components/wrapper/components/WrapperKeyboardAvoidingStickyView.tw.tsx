import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingStickyViewPropsTw } from '@types';

import { cn, customConsole } from '@utils';

import { Wrapper } from './Wrapper.tw';
import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.tw';

// Optional import with error handling
let KeyboardStickyView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardStickyView = keyboardController.KeyboardStickyView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * KeyboardAvoidingView wrapper component for keyboard avoidance.
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 *
 * // With SafeAreaView
 * <WrapperKeyboardAvoidingView hasSafeArea edges={['top', 'bottom']}>
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 * ```
 * @param className - Custom CSS classes for the wrapper
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param px - Padding for the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param edges - Safe area edges to apply when hasSafeArea is true
 * @param style - Style prop
 * @param enabled - Whether the keyboard avoiding view is enabled
 * @param offset - Offset for the keyboard avoiding sticky view { close: number; open: number }
 * @param children - Child components to render
 */
export function WrapperKeyboardAvoidingStickyView(props: PropsWithChildren<WrapperKeyboardAvoidingStickyViewPropsTw>) {
  const { children, className, edges, fill, grow, hasSafeArea, px, style, ...rest } = props;

  if (!KeyboardStickyView) {
    return (
      <Wrapper
        className={cn(wrapperTv({ fill, grow, px }), className)}
        hasSafeArea={hasSafeArea}
        edges={edges}
        {...rest}
      >
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardStickyView className={cn(wrapperTv({ fill, grow, px }), className)} {...rest}>
      {children}
    </KeyboardStickyView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingStickyView.displayName = 'WrapperKeyboardAvoidingStickyView';
