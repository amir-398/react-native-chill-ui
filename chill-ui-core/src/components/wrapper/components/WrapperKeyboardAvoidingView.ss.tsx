import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingViewPropsSs } from '@types';

import { Wrapper } from './Wrapper.ss';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { wrapperDefaultProps } from '../utils/defaultProps';

// Optional import with error handling
let KeyboardAvoidingView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardAvoidingView = keyboardController.KeyboardAvoidingView;
  }
} catch {
  console.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * KeyboardAvoidingView wrapper component for keyboard avoidance.
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 * ```
 *
 * @param fill - Whether to fill the wrapper
 * @param px - Padding for the wrapper
 * @param style - Style prop
 * @param keyboardVerticalOffset - Keyboard vertical offset
 * @param behavior - Behavior of the keyboard avoiding view
 * @param enabled - Whether the keyboard avoiding view is enabled
 * @param contentContainerStyle - Content container style
 * @param children - Child components to render
 */
export function WrapperKeyboardAvoidingView(props: PropsWithChildren<WrapperKeyboardAvoidingViewPropsSs>) {
  const {
    children,
    fill,
    keyboardVerticalOffset = wrapperDefaultProps.keyboardVerticalOffset,
    px,
    style,
    ...rest
  } = props;

  if (!KeyboardAvoidingView) {
    console.error(
      'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
    );
    return (
      <Wrapper style={[wrapperSv({ fill, px }), style]} {...rest}>
        {children}
      </Wrapper>
    );
  }

  return (
    <KeyboardAvoidingView
      style={[wrapperSv({ fill, px }), style]}
      keyboardVerticalOffset={keyboardVerticalOffset}
      {...rest}
    >
      {children}
    </KeyboardAvoidingView>
  );
}

WrapperKeyboardAvoidingView.displayName = 'WrapperKeyboardAvoidingView';
