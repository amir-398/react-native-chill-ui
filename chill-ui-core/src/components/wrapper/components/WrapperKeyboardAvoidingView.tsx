import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingViewPropsTw } from '@types';

import { classNameHandler, cn, customConsole, styleHandler } from '@utils';

import { Wrapper } from './Wrapper';
import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { wrapperDefaultProps } from '../utils/defaultProps';
import { WrapperSafeAreaView } from './WrapperSafeAreaView';

// Optional import with error handling
let KeyboardAvoidingView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardAvoidingView = keyboardController.KeyboardAvoidingView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * The `<WrapperKeyboardAvoidingView />` component provides a KeyboardAvoidingView wrapper for keyboard avoidance.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { WrapperKeyboardAvoidingView } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 * ```
 *
 * @param behavior - Behavior of the keyboard avoiding view: `'height'` | `'position'` | `'padding'` | `'translate-with-padding'` (default: `'padding'`)
 * @param className - Custom CSS classes for the wrapper (NativeWind only)
 * @param contentContainerStyle - Content container style when behavior is position
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param enabled - Whether the keyboard avoiding view is enabled (default: `true`)
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param keyboardVerticalOffset - Keyboard vertical offset (default: `0`)
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'` | `'2xl'` | `'3xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function WrapperKeyboardAvoidingView(props: PropsWithChildren<WrapperKeyboardAvoidingViewPropsTw>) {
  const {
    children,
    className,
    edges,
    fill,
    grow,
    hasSafeArea,
    keyboardVerticalOffset = wrapperDefaultProps.keyboardVerticalOffset,
    px,
    style,
    ...rest
  } = props;

  if (!KeyboardAvoidingView) {
    return (
      <Wrapper
        {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
        {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
        hasSafeArea={hasSafeArea}
        edges={edges}
        {...rest}
      >
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardAvoidingView
      {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
      {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
      behavior={wrapperDefaultProps.behavior}
      keyboardVerticalOffset={keyboardVerticalOffset}
      {...rest}
    >
      {children}
    </KeyboardAvoidingView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingView.displayName = 'WrapperKeyboardAvoidingView';
