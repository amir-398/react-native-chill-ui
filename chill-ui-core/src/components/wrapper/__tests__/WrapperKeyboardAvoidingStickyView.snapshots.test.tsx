import React from 'react';
import { render } from '@testing-library/react-native';
import { WrapperKeyboardAvoidingStickyView } from '../components/WrapperKeyboardAvoidingStickyView';

describe('WrapperKeyboardAvoidingStickyView Snapshots', () => {
  it('should match snapshot with basic props', () => {
    const tree = render(<WrapperKeyboardAvoidingStickyView>Content</WrapperKeyboardAvoidingStickyView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with fill prop', () => {
    const tree = render(<WrapperKeyboardAvoidingStickyView fill>Content</WrapperKeyboardAvoidingStickyView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with grow prop', () => {
    const tree = render(<WrapperKeyboardAvoidingStickyView grow>Content</WrapperKeyboardAvoidingStickyView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with px prop', () => {
    const tree = render(<WrapperKeyboardAvoidingStickyView px="md">Content</WrapperKeyboardAvoidingStickyView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom className', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView className="custom-class">Content</WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom style', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView style={{ marginBottom: 8 }}>Content</WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with enabled prop', () => {
    const tree = render(<WrapperKeyboardAvoidingStickyView enabled={false}>Content</WrapperKeyboardAvoidingStickyView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with offset prop', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView offset={{ close: 0, open: 20 }}>Content</WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with hasSafeArea prop', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView hasSafeArea edges={['top', 'bottom']}>
        Content
      </WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with edges prop', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView hasSafeArea edges={['top', 'left', 'right', 'bottom']}>
        Content
      </WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with accessibility props', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView
        accessible
        accessibilityLabel="Keyboard avoiding sticky container"
        testID="keyboard-avoiding-sticky-view"
      >
        Content
      </WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with all props combined', () => {
    const tree = render(
      <WrapperKeyboardAvoidingStickyView
        fill
        grow
        px="lg"
        className="custom-class"
        style={{ marginBottom: 16 }}
        enabled={true}
        offset={{ close: 0, open: 20 }}
        hasSafeArea
        edges={['top', 'bottom']}
        testID="keyboard-avoiding-sticky-view"
        accessible
        accessibilityLabel="Custom keyboard avoiding sticky view"
      >
        Content
      </WrapperKeyboardAvoidingStickyView>,
    );
    expect(tree).toMatchSnapshot();
  });
});
