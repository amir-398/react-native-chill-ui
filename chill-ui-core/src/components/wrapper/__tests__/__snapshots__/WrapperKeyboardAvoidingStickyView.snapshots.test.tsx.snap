// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with accessibility props 1`] = `
<View
  accessibilityLabel="Keyboard avoiding sticky container"
  accessible={true}
  className="flex-1 px-[16px]"
  testID="keyboard-avoiding-sticky-view"
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with all props combined 1`] = `
<RNCSafeAreaView
  className="flex-1 px-0"
  edges={
    {
      "bottom": "additive",
      "left": "off",
      "right": "off",
      "top": "additive",
    }
  }
>
  <View
    accessibilityLabel="Custom keyboard avoiding sticky view"
    accessible={true}
    className="flex-grow px-[24px] flex-grow flex-none custom-class"
    enabled={true}
    offset={
      {
        "close": 0,
        "open": 20,
      }
    }
    style={
      {
        "marginBottom": 16,
      }
    }
    testID="keyboard-avoiding-sticky-view"
  >
    Content
  </View>
</RNCSafeAreaView>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with basic props 1`] = `
<View
  className="flex-1 px-[16px]"
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with custom className 1`] = `
<View
  className="flex-1 px-[16px] custom-class"
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with custom style 1`] = `
<View
  className="flex-1 px-[16px]"
  style={
    {
      "marginBottom": 8,
    }
  }
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with edges prop 1`] = `
<RNCSafeAreaView
  className="flex-1 px-0"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    className="flex-1 px-[16px]"
  >
    Content
  </View>
</RNCSafeAreaView>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with enabled prop 1`] = `
<View
  className="flex-1 px-[16px]"
  enabled={false}
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with fill prop 1`] = `
<View
  className="flex-1 px-[16px]"
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with grow prop 1`] = `
<View
  className="flex-grow px-[16px] flex-grow flex-none"
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with hasSafeArea prop 1`] = `
<RNCSafeAreaView
  className="flex-1 px-0"
  edges={
    {
      "bottom": "additive",
      "left": "off",
      "right": "off",
      "top": "additive",
    }
  }
>
  <View
    className="flex-1 px-[16px]"
  >
    Content
  </View>
</RNCSafeAreaView>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with offset prop 1`] = `
<View
  className="flex-1 px-[16px]"
  offset={
    {
      "close": 0,
      "open": 20,
    }
  }
>
  Content
</View>
`;

exports[`WrapperKeyboardAvoidingStickyView Snapshots should match snapshot with px prop 1`] = `
<View
  className="flex-1 px-[16px]"
>
  Content
</View>
`;
